sdcm/cluster.py

```diff
@@ -2606,17 +2606,28 @@ def prepare_files_for_archive(self, fileslist):
                 self.remoter.run('cp -r %s %s' % (f, new_full_path), ignore_status=True)
         return root_dir
 
-    def generate_coredump_file(self, restart_scylla=True, node_name: str = None, message: str = None):
+    def generate_coredump_file(self, restart_scylla=True, node_name: str = None, message: str = None, generate_diagnosis=False):
         message = message or "Generating a Scylla core dump by SIGQUIT..\n"
         if node_name:
             message += f"Node: {node_name}"
-        InfoEvent(severity=Severity.ERROR, message=message)
-        self.remoter.sudo("pkill -f --signal 3 /usr/bin/scylla")
-        self.wait_db_down(timeout=600)
-        if restart_scylla:
-            self.log.debug('Restart scylla server')
-            self.stop_scylla(timeout=600)
-            self.start_scylla(timeout=600)
+        if generate_diagnosis:
+            InfoEvent(severity=Severity.DEBUG, message=message)
+            with DbNodeLogger([self], "send SIGQUIT to scylla", target_node=self, additional_info="diagnosis dump" if generate_diagnosis else "coredump"):
+                self.remoter.sudo("pkill -3 scylla")
```

Comment:
`pkill -f --signal 3 /usr/bin/scylla` used below (formerly) doesn't work? Maybe we should use the same for both (can be new one if it works better)

Besides I don't think we should differentiate both cases - we should always return cluster to base state (all nodes runnig) - there could be other validators that require scylla to be up.